serde_derive/src/internals/attr.rs

```diff
@@ -341,6 +356,43 @@ impl Container {
                             }
                         }
                     }
+                } else if meta.path == RENAME_ALL_FIELDS {
+                    // #[serde(rename_all_fields = "foo")]
+                    // #[serde(rename_all_fields(serialize = "foo", deserialize = "bar"))]
+                    match item.data {
+                        syn::Data::Enum(_) => {
+                            let one_name = meta.input.peek(Token![=]);
+                            let (ser, de) = get_renames(cx, RENAME_ALL, &meta)?;
+                            if let Some(ser) = ser {
+                                match RenameRule::from_str(&ser.value()) {
+                                    Ok(rename_rule) => {
+                                        rename_all_fields_ser_rule.set(&meta.path, rename_rule);
+                                    }
+                                    Err(err) => cx.error_spanned_by(ser, err),
+                                }
+                            }
+                            if let Some(de) = de {
+                                match RenameRule::from_str(&de.value()) {
+                                    Ok(rename_rule) => {
+                                        rename_all_fields_de_rule.set(&meta.path, rename_rule);
+                                    }
+                                    Err(err) => {
+                                        if !one_name {
+                                            cx.error_spanned_by(de, err);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        syn::Data::Struct(syn::DataStruct { struct_token, .. }) => {
+                            let msg = "#[serde(rename_all_fields)] can only be used on enums";
+                            cx.error_spanned_by(struct_token, msg);
+                        }
+                        syn::Data::Union(syn::DataUnion { union_token, .. }) => {
+                            let msg = "#[serde(rename_all_fields)] can only be used on enums";
+                            cx.error_spanned_by(union_token, msg);
+                        }
```

Comment:
I think using the `struct`/`union` token for the span of this error is not a good choice. The problem is not that the user wrote a struct. The problem is that the user wrote a `rename_all_fields` attribute on their struct. The fix is not for the user to turn their struct into an enum. The fix is to change the attribute. So the error should point to the attribute.

Reply:
Makes sense. When I just updated this, I noticed the (container-level) `#[serde(default)]` does the same thing of pointing to the enum / struct token. Should I send another PR to change that?

Reply:
Yep, let's fix that too. Good catch.